### PR TITLE
Remove buildspec in task spec

### LINKED ISSUE #228: Move `buildSpec` up a level inside Task.spec
## Expected Behavior
`Task.spec` should no longer contain a `Build.spec`. All fields that we want to support from `Build` in `Task` should be moved up one level and explicitly included inside `Task.spec`, e.g.:

```yaml
spec:
      steps:
          - name: runMake
            image: ubuntu
            command: ['make']
            args: ['${inputs.params.makeTarget}']
```

## Actual Behavior
`Task.spec` contains `Build.spec`, however we limit the fields which can be used, for example we do not allow the user to provide a reference to a Build template:

https://github.com/knative/build-pipeline/blob/df83248d5bc51103288d3704cc53d7afcf852e63/pkg/apis/pipeline/v1alpha1/task_validation.go#L47-L50

Tasks currently contain a `buildSpec` field, e.g.:

```yaml
spec:
    buildSpec:
        steps:
            - name: runMake
              image: ubuntu
              command: ['make']
              args: ['${inputs.params.makeTarget}']
```

## Steps to Reproduce the Problem
n/a

## Additional Info
Part of #147

### PR DESCRIPTION
**Pull request by shashwathi**
Fixes #228 

- This is the first step towards consuming build into build pipeline

TODO
- [x] Use `sources` instead of source

### CODE REVIEW

**Comment by shashwathi** [comment]
/hold 

Waiting for #232 to merged first

File: pkg/apis/pipeline/v1alpha1/task_validation_test.go [hunk 1]
```diff
@@ -201,28 +197,16 @@ func TestTaskSpec_ValidateError(t *testing.T) {
...
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			ts := &TaskSpec{
-				Inputs:    tt.fields.Inputs,
-				Outputs:   tt.fields.Outputs,
-				BuildSpec: tt.fields.BuildSpec,
+				Inputs:  tt.fields.Inputs,
+				Outputs: tt.fields.Outputs,
+				Steps:   tt.fields.BuildSteps,
```
[the remark this answers is not in the document]
Comment (shashwathi) [reply pkg/apis/pipeline/v1alpha1/task_validation_test.go]: In this test there are use cases where buildstep is not set or empty list of containers. In this line it makes sense to use `BuildSteps` field.

**Comment by shashwathi** [comment]
@ImJasonH : I have addressed your comments.

I have also removed `hold` as `Source` is not supported anymore and `Sources` is used from recent build release.

**Comment by tejal29** [comment]
Wov! Great to see this in!

**Comment by bobcatfish** [comment]
very cool @shashwathi !! Thanks!! :tada:

**Pull request merged**